lib/ansible/cli/doc.py

```diff
@@ -322,123 +330,108 @@ def add_fields(self, text, fields, limit, opt_indent):
                 del opt['choices']
             default = ''
             if 'default' in opt or not required:
-                default = "[Default: " + str(opt.pop('default', '(null)')) + "]"
+                default = "[Default: %s" % str(opt.pop('default', '(null)')) + "]"
+
             text.append(textwrap.fill(CLI.tty_ify(aliases + choices + default), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
 
             if 'options' in opt:
-                text.append(opt_indent + "options:\n")
-                self.add_fields(text, opt['options'], limit, opt_indent + opt_indent)
-                text.append('')
-                del opt['options']
+                text.append("%soptions:\n" % opt_indent)
+                self.add_fields(text, opt.pop('options'), limit, opt_indent + opt_indent)
 
             if 'spec' in opt:
-                text.append(opt_indent + "spec:\n")
-                self.add_fields(text, opt['spec'], limit, opt_indent + opt_indent)
-                text.append('')
-                del opt['spec']
-
-            for conf in ('config', 'env_vars', 'host_vars'):
-                if conf in opt:
-                    text.append(textwrap.fill(CLI.tty_ify("%s: " % conf), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
-                    for entry in opt[conf]:
-                        if isinstance(entry, dict):
-                            pre = "  -"
-                            for key in entry:
-                                text.append(textwrap.fill(CLI.tty_ify("%s %s: %s" % (pre, key, entry[key])),
-                                            limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
-                                pre = "   "
-                        else:
-                            text.append(textwrap.fill(CLI.tty_ify("  - %s" % entry), limit, initial_indent=opt_indent, subsequent_indent=opt_indent))
-                    del opt[conf]
+                text.append("%sspec:\n" % opt_indent)
+                self.add_fields(text, opt.pop('spec'), limit, opt_indent + opt_indent)
+
+            conf = {}
+            for config in ('env', 'ini', 'yaml', 'vars'):
+                if config in opt and opt[config]:
+                    conf[config] = opt.pop(config)
+            else:
```

Comment:
`else` seems useless here, it could be removed (then `if` block below must be moved at the same level than the `for` loop).